src/ray/util/util.h

```diff
@@ -331,54 +332,31 @@ class ThreadPrivate {
   explicit ThreadPrivate(Ts &&...ts) : t_(std::forward<Ts>(ts)...) {}
 
   T &operator*() {
-    ThreadCheck();
+    RAY_CHECK(thread_checker_.IsOnSameThread());
     return t_;
   }
 
   T *operator->() {
-    ThreadCheck();
+    RAY_CHECK(thread_checker_.IsOnSameThread());
     return &t_;
   }
 
   const T &operator*() const {
-    ThreadCheck();
+    RAY_CHECK(thread_checker_.IsOnSameThread());
     return t_;
   }
 
   const T *operator->() const {
-    ThreadCheck();
+    RAY_CHECK(thread_checker_.IsOnSameThread());
     return &t_;
   }
 
  private:
-  void ThreadCheck() const {
-    // ThreadCheck is not a thread safe function and at the same time, multiple
-    // threads might be accessing id_ at the same time.
-    // Here we only introduce mutex to protect write instead of read for the
-    // following reasons:
-    //    - read and write at the same time for `id_` is fine since this is a
-    //      trivial object. And since we are using this to detect errors,
-    //      it doesn't matter which value it is.
-    //    - read and write of `thread_name_` is not good. But it will only be
-    //      read when we crash the program.
-    //
-    if (id_ == std::thread::id()) {
-      // Protect thread_name_
-      std::lock_guard<std::mutex> _(mutex_);
-      thread_name_ = GetThreadName();
-      RAY_LOG(DEBUG) << "First accessed in thread " << thread_name_;
-      id_ = std::this_thread::get_id();
-    }
-
-    RAY_CHECK(id_ == std::this_thread::get_id())
-        << "A variable private to thread " << thread_name_ << " was accessed in thread "
-        << GetThreadName();
-  }
-
   T t_;
   mutable std::string thread_name_;
   mutable std::thread::id id_;
```

Comment:
pls also remove this. Maybe also thread_name_ and mutex_ ?

Reply:
Makes sense, updated.